File: web3_test_connect.js
const Web3 = require('web3').default;
import {INFURA_API_KEY} from "@env"

const url = `https://sepolia.infura.io/v3/${INFURA_API_KEY}`;

const web3 = new Web3(url);

web3.eth.net.isListening()
    .then(() => console.log('Connected to Sepolia Testnet'))
    .catch(e => console.log('Failed to connect: ', e.message));


const axios = require('axios');
import {INFURA_API_KEY} from "@env"

const infuraUrl = `https://goerli.infura.io/v3/${INFURA_API_KEY}`;

const data = JSON.stringify({
  jsonrpc: '2.0',
  method: 'eth_blockNumber',
  params: [],
  id: 1,
});

const config = {
  method: 'post',
  url: infuraUrl,
  headers: { 
    'Content-Type': 'application/json',
  },
  data : data
};

axios(config)
.then(function (response) {
  console.log(JSON.stringify(response.data));
})
.catch(function (error) {
  console.error(error);
});

File: web3_test_connect.py
from web3 import Web3
from dotenv import load_dotenv
import os
load_dotenv()
ALCHEMY_API_KEY = os.getenv("ALCHEMY_API_KEY")
INFURA_API_KEY = os.getenv("INFURA_API_KEY")
print(f"ALCHEMY_API_KEY: {ALCHEMY_API_KEY}")
# infura sepolia testnet dns error, use goerli testnet or use Alchemy sepolia testnet
url = f'https://goerli.infura.io/v3/{INFURA_API_KEY}'
alchemy_url = f"https://eth-sepolia.g.alchemy.com/v2/{ALCHEMY_API_KEY}"

# # w3 = Web3(Web3.HTTPProvider(url))
w3 = Web3(Web3.HTTPProvider(alchemy_url))

if w3.is_connected():
    print("Connected to alchemy sepolia Testnet")
else:
    print("Failed to connect")
